wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -137,17 +145,26 @@ sendMoney
     -> NonEmpty (CId Addr, Coin)
     -> m CTx
 sendMoney SendActions{..} passphrase moneySource dstDistr = do
-    addrMetas' <- getMoneySourceAddresses moneySource
+    let srcWallet = getMoneySourceWallet moneySource
+    rootSk <- getSKById srcWallet
+    checkPassMatches passphrase rootSk `whenNothing`
+        throwM (RequestError "Passphrase doesn't match")
+
+    addrMetas'' <- getMoneySourceAddresses moneySource
+    utxo <- getCachedUtxo
+    let addrsWithMoney =
+            HS.fromList . map (addressToCId . txOutAddress . toaOut) $
+            toList utxo
+        addrMetas' = filter (flip HS.member addrsWithMoney . cwamId) addrMetas''
+
     addrMetas <- nonEmpty addrMetas' `whenNothing`
-        throwM (RequestError "Given money source has no addresses!")
+        throwM (RequestError "Given money source has no addresses with money!")
```

Comment:
If this fix will be used not only by bittrex, we have to ensure this error can be handled by daedalus

Reply:
https://issues.serokell.io/issue/DAEF-511